src/parser.ts

```diff
@@ -2583,12 +2583,34 @@ export class Parser {
     parseThrowStatement(): Node.ThrowStatement {
         const node = this.createNode();
         this.expectKeyword('throw');
+        const hasLineTerminator = this.hasLineTerminator;
+        const source = this.scanner.source;
 
-        if (this.hasLineTerminator) {
-            this.throwError(Messages.NewlineAfterThrow);
+        let argument;
+        try {
+            argument = this.parseExpression();
+        } catch (e) {
+            if (!hasLineTerminator) {
+                throw e;
+            }
+        }
+
+        if (hasLineTerminator) {
+            const isTemplateLiteral =
+                argument && argument.type === 'TemplateLiteral';
+            const hasLineTerminatorBeforeTemplate =
+                isTemplateLiteral &&
+                source.trim().indexOf('\n') < source.trim().indexOf('`');
```

Comment:
This is almost guaranteed to be wrong, `source` holds the _entire_ input to parsed. You can't just run `trim` and look for a particular character. What if there are multiple template literals in the input?